Source/Core/Common/bbaserver.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "bbaserver.h"
+
+#include <QTcpServer>
+#include <QHostAddress>
+#include <QNetworkProxy>
+#include <QTimerEvent>
+
+#include "bbaclient.h"
+#include "bbadebug.h"
+
+BbaServer::BbaServer(QObject *parent) :
+  QObject(parent),
+  m_server(*new QTcpServer(this))
+{
+  connect(&m_server, &QTcpServer::newConnection, this, &BbaServer::newConnection);
+}
+
+bool BbaServer::listen(const QHostAddress &address, quint16 port)
+{
+  const auto result = m_server.listen(address, port);
+  if (result)
+    logInfo() << "Started listening on" << address << port;
+  else
+    logInfo() << "Could not start listening on" << address << port << "because" << m_server.errorString();
+  return result;
+}
+
+bool BbaServer::listen(const QString &address, quint16 port)
+{
+  return listen(address.isEmpty() ? QHostAddress::Any : QHostAddress(address), port);
+}
+
+void BbaServer::close()
+{
+  logInfo() << "Stopped listening";
+  return m_server.close();
+}
+
+bool BbaServer::isListening() const
+{
+  return m_server.isListening();
+}
+
+void BbaServer::setMaxPendingConnections(int numConnections)
+{
+  m_server.setMaxPendingConnections(numConnections);
+}
+
+int BbaServer::maxPendingConnections() const
+{
+  return m_server.maxPendingConnections();
+}
+
+quint16 BbaServer::serverPort() const
+{
+  return m_server.serverPort();
+}
+
+QHostAddress BbaServer::serverAddress() const
+{
+  return m_server.serverAddress();
+}
+
+//QAbstractSocket::SocketError BbaServer::serverError() const
+//{
+//  return m_server.serverError();
+//}
+
+QString BbaServer::errorString() const
+{
+  return m_server.errorString();
+}
+
+void BbaServer::pauseAccepting()
+{
+  m_server.pauseAccepting();
+}
+
+void BbaServer::resumeAccepting()
+{
+  m_server.resumeAccepting();
+}
+
+#ifndef QT_NO_NETWORKPROXY
+void BbaServer::setProxy(const QNetworkProxy &networkProxy)
+{
+  m_server.setProxy(networkProxy);
+}
+
+QNetworkProxy BbaServer::proxy() const
+{
+  return m_server.proxy();
+}
+#endif
+
+BbaDebug BbaServer::logInfo()
+{
+  return BbaDebug(*this);
+}
+
+void BbaServer::timerEvent(QTimerEvent *ev)
+{
+  if (ev->timerId() == m_timerId)
+  {
+    if (m_counter > 0)
+    {
+      logInfo() << m_counter << "packets transmitted";
+      m_counter = 0;
+    }
+  }
+  else
+    QObject::timerEvent(ev);
+}
+
+void BbaServer::newConnection()
+{
+  if (auto socket = m_server.nextPendingConnection())
+  {
+    auto client = new BbaClient(*socket, *this);
+    for(auto otherClient : m_clients)
```

Comment:
All variables should be using `snake_case` formatting for their names.

```suggestion
    for (auto* other_client : m_clients)
```